force-app/main/default/classes/CRM_JournalUtilities.cls

```diff
@@ -42,65 +42,49 @@ public class CRM_JournalUtilities {
-            if (existingJEntries.size() > 0) {
-                resp.success = true;
-                resp.duplicate = existingJEntries[0].Id != referenceId;
-                Boolean journalCompleted = (Boolean) responseMap.get('journalpostferdigstilt');
-                resp.journalPostStatus = journalCompleted ? 'Completed' : 'In Progress';
-            } else {
-                resp.success = false;
-                LoggerUtility logger = new LoggerUtility('Journal');
-                SObject objectRef = String.isNotBlank(referenceId)
-                    ? referenceId.getSobjectType().newSObject(referenceId)
-                    : null;
-
-                logger.httpError(
-                    'Error posting journal entry, got conflict but found no existing journal post',
-                    httpResp,
-                    objectRef,
-                    CRM_ApplicationDomain.Domain.NKS
-                );
-                String errorRef = logger.peek().UUID__c; //Unique error reference for troubleshooting
-                logger.publish();
-                resp.errorMessage = errorRef;
+            if (httpResp.getStatusCode() == 409) {
+                List<Journal_Entry__c> existingJEntries = [
+                    SELECT CRM_Thread__c, CRM_Conversation_Note__c
+                    FROM Journal_Entry__c
+                    WHERE Journal_Entry_Id__c = :resp.journalpostId
+                    LIMIT 1
+                ];
+                resp.duplicate =
+                    existingJEntries.size() > 0 &&
+                    (existingJEntries[0].CRM_Thread__c == referenceId ||
+                    existingJEntries[0].CRM_Conversation_Note__c == referenceId);
             }
         } else {
             resp.success = false;
-            //Creates a log entry for troubleshooting
-            LoggerUtility logger = new LoggerUtility('Journal');
-            SObject objectRef = String.isNotBlank(referenceId)
-                ? referenceId.getSobjectType().newSObject(referenceId)
-                : null;
-
-            logger.httpError('Error posting journal entry', httpResp, objectRef, CRM_ApplicationDomain.Domain.NKS);
-            String errorRef = logger.peek().UUID__c; //Unique error reference for troubleshooting
-            logger.publish();
-            resp.errorMessage = errorRef;
+            handleLogging(httpResp, referenceId, resp);
         }
 
         return resp;
     }
 
+    private static void handleLogging(HttpResponse httpResp, Id referenceId, UtilityResponse resp) {
+        LoggerUtility logger = new LoggerUtility('Journal');
+        SObject objectRef = String.isNotBlank(referenceId)
+            ? referenceId.getSobjectType().newSObject(referenceId)
+            : null;
+        logger.httpError('Error posting journal entry', httpResp, objectRef, CRM_ApplicationDomain.Domain.NKS);
+        String errorRef = logger.peek().UUID__c;
+        logger.publish();
+        resp.errorMessage = errorRef;
+    }
+
```

Comment:
Jeg snakket nylig med Eivind om dette faktisk, og han pekte på at det er litt dumt å ha en egen "log" metode i klassene våre, fordi da vil ikke source_function delen av ApplicationLog gi noe verdi lenger. Vi burde enten slutte å bruke de (for det meste), eller utvide logge rammeverket litt til å håndtere slike metoder bedre.